### ISSUE TITLE
[Perf] Home screen is refreshed whenever a note is (auto-)saved

### ISSUE THREAD

**Issue by adil192** [adil192/saber#650]
### Motivation

A note is auto-saved (after a small delay) when it is edited. The home screen watches for file changes and reloads notes that have changed. Therefore, after every change to a note, it is saved to disk, then read back from disk, parsed, and redisplayed on a screen that isn't even visible to the user!

I've noticed that if I use a large PDF (e.g. 40 pages) the editor is considerably laggy, but if I split that same PDF into individual chapters (say about 5-10 pages each) and only import one chapter into a note, there is no noticible lag.

### Solution

If we only reload the home screen preview of the note once we return to the home screen, rather than immediately after a note is saved, it might aleviate some of this lag.

There is also merit in looking into optimising the saving process, but that is a separate issue.

### Saber version

v0.12.5 (12050)

### Device

- Device: Google Pixel 6 Pro
- OS: Android 13

### Anything else?

To the end-user, this lag in saving and reloading a file will probably be mostly alleviated after #432 is done. But it's still a good optimisation to have.
